lib/core/network/api/services/token_service.dart

```diff
@@ -21,8 +23,21 @@ class TokenService {
         "code": "$tokenContract",
         "symbol": "$symbol",
       }''';
-      final res = await networkingManager.post(Endpoints.getCurrencyBalance, data: requestBody);
-      final tokenString = res.data[0];
+      final Response<List> res = await networkingManager.post<List>(
+        Endpoints.getCurrencyBalance,
+        data: requestBody,
+      );
+
+      if(res.data == null) {
```

Comment:
seems like the response is null if the user never had the token. 

Null then becomes 0